### ISSUE TITLE
Prepare data for "get reference" functions 

### ISSUE THREAD

**Issue by valegagge** [robotology/icub-firmware#53]
This issue is related to https://github.com/robotology/icub-main/issues/264.

Fw should fill target fields in joint status structure.

**Comment by valegagge** [comment]
Speaking with @barbalberto and @randaz81, we decided to add a field for taget position configured by IPositionDirect::setPosition. therefore new strucutre for target is:
```
typedef struct                  // size is:  4+4+4+4+4+4 = 24
{
    eOmeas_position_t           trgt_position;              /**< the target position of the joint */           
    eOmeas_position_t           trgt_positionraw;           /**< the target position raw of the joint */           
    eOmeas_velocity_t           trgt_velocity;              /**< the target velocity of the joint */          
    eOmeas_acceleration_t       trgt_acceleration;          /**< the target acceleration of the joint */       
    eOmeas_torque_t             trgt_torque;                /**< the target torque of the joint */
    int32_t                     trgt_openloop;              /**< the target openloop of the joint */
} eOmc_joint_status_target_t;   EO_VERIFYsizeof(eOmc_joint_status_target_t, 24); 
```
In this way each control mode interface has each target.

**Comment by valegagge** [comment]
almost done.
Currently targets are updated in setpoint command's callback.
In new motor controller we should updated target values by eo_emsController_GetJointStatus

**Comment by valegagge** [comment]
Tested with movementReferencesTest in icub-test.
Results:
- on mc4plus and ems with 2foc ok
- on ems-with-mc4 nok, but it is not a bug. 
  We decided that references aren't asked to mc4 but cached in ems, so in some particular situation the given reference is not correct. For example if the joint j is in openloop control mode and the user calls PositionMove(j, mypos) the getreference function returns mypos instead of the last position given when j was in controlled in position.

We chose to implement this behaviour to reduce implementation time. Maybe in the future we may make all boards' behaviour uniform.
